packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -146,7 +163,9 @@ library LibProving {
         fc.provenAt = uint64(block.timestamp);
         fc.prover = evidence.prover;
 
-        if (evidence.prover != address(0)) {
+        // evidence.prover cannot be address(0) at this point -
+        // either system or a regular prover
```

Comment:
Why not? Seems to get there if the fc is being overwritten by the oracle prover or do I misunderstand the comment?

Reply:
Old commenting from prev. version, thanks for the note!

Reply:
Still gotta skip the proof verification below if `&& evidence.prover != address(0)` as well no?

Reply:
Yes, fixed.